pyrobosim/pyrobosim/core/yaml.py

```diff
@@ -175,39 +175,44 @@ def add_global_path_planner(self):
         ):
             return
+
         planner_data = self.data["global_path_planner"]
         planner_type = planner_data["type"]
+        create_planner = False
 
         if planner_type == "search_graph":
-            max_edge_dist = planner_data.get("max_edge_dist", np.inf)
-            collision_check_dist = planner_data.get("collision_check_dist", 0.1)
-            self.world.create_search_graph(
-                max_edge_dist=max_edge_dist,
-                collision_check_dist=collision_check_dist,
-                create_planner=True,
-            )
-        else:
-            # Always make a search graph as we use it for other things.
-            max_edge_dist = planner_data.get("max_edge_dist", np.inf)
-            collision_check_dist = planner_data.get("collision_check_dist", 0.1)
-            self.world.create_search_graph(
-                max_edge_dist=max_edge_dist,
-                collision_check_dist=collision_check_dist,
-                create_planner=False,
+            create_planner = True
+
+        elif planner_type == "prm":
+            from pyrobosim.navigation.prm import PRMPlanner
+
+            max_nodes = planner_data.get("max_nodes", 100)
+            max_connection_dist = planner_data.get("max_connection_dist", 1.0)
+            self.world.path_planner = PRMPlanner(
+                self.world,
+                max_nodes=max_nodes,
+                max_connection_dist=max_connection_dist,
             )
 
-            if planner_type == "prm":
-                from pyrobosim.navigation.prm import PRMPlanner
+        elif planner_type == "astar_grid":
+            from pyrobosim.navigation.astar_grid import AStarGridPlanner
 
-                max_nodes = planner_data.get("max_nodes", 100)
-                max_connection_dist = planner_data.get("max_connection_dist", 1.0)
-                self.world.path_planner = PRMPlanner(
-                    self.world,
-                    max_nodes=max_nodes,
-                    max_connection_dist=max_connection_dist,
-                )
-            else:
-                warnings.warn(f"Invalid global planner type specified: {planner_type}")
+            resolution = planner_data.get("resolution", 0.05)
+            inflation_radius = planner_data.get("inflation_radius", 0.1)
+            self.world.path_planner = AStarGridPlanner(
+                self.world, resolution=resolution, inflation_radius=inflation_radius
+            )
+        else:
+            warnings.warn(f"Invalid global planner type specified: {planner_type}")
+
+        # Always make a search graph as we use it for other things.
+        max_edge_dist = planner_data.get("max_edge_dist", np.inf)
+        collision_check_dist = planner_data.get("collision_check_dist", 0.1)
+        self.world.create_search_graph(
+            max_edge_dist=max_edge_dist,
+            collision_check_dist=collision_check_dist,
+            create_planner=create_planner,
```

Comment:
Nice cleanup -- Took me a while to figure it out, but yeah, looks good!

Reply:
I noticed that only the `create_planner` was differing in the `if-else` block.   So I refactored it to such that only `create_planner` is set in the conditionals.  And put the search graph creation at the end since we have to create a graph anyway.